The runtime/main build is broken with cmake 3.6.2

Back in https://github.com/dotnet/runtime/pull/39044 we made sure cmake 3.6.2 was supported for building dotnet/runtime.

Unfortunately, https://github.com/dotnet/runtime/pull/55370, seems to have broken that ability:

```
  -- Performing Test HAVE_IN_EXCL_UNLINK
  -- Performing Test HAVE_IN_EXCL_UNLINK - Success
  -- Performing Test HAVE_TCP_H_TCP_KEEPALIVE
  -- Performing Test HAVE_TCP_H_TCP_KEEPALIVE - Failed
  -- Performing Test HAVE_BUILTIN_MUL_OVERFLOW
  -- Performing Test HAVE_BUILTIN_MUL_OVERFLOW - Success
  -- Performing Test HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS
  -- Performing Test HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS - Failed
  -- Looking for ucol_setMaxVariable
  -- Looking for ucol_setMaxVariable - not found
  CMake Error at /home/tester/runtime/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt:64 (add_link_options):
    Unknown CMake command "add_link_options".
  
  
  -- Configuring incomplete, errors occurred!
```

### Configuration

This was the result of trying to build commit 3150bfa50618cc98f546ffae7652b12c17cea702 on RHEL 7 x64 with `llvm-toolset-10.0-cmake.x86_64 0:3.6.2-12.el7`

### Other information

We decided to stick with CMake 3.6.2 for 5.0. Maybe we are okay with changing that dependency and asking for a higher version of CMake for 6.0? But if we do that, then I think that should be an explicit decision and announced, like suggested by https://github.com/dotnet/designs/blob/main/accepted/2021/platform-dependencies/platform-dependencies.md

--- a/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
+++ b/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
@@ -62,7 +62,7 @@ if (FEATURE_DISTRO_AGNOSTIC_SSL)
     )
     add_definitions(-DFEATURE_DISTRO_AGNOSTIC_SSL)
     add_compile_options(-pthread)
-    add_link_options(-pthread)
+    add_linker_flag(-pthread)
 endif()
 
 add_library(objlib OBJECT ${NATIVECRYPTO_SOURCES} ${VERSION_FILE_PATH})

